### PR TITLE
[merged] Fix proxy authentication

### PR DESCRIPTION
**Pull request by jlebon**
Make sure to pass the proxy authentication if asked for it. Though libsoup should do that by default (see https://bugzilla.gnome.org/show_bug.cgi?id=772932).

Also includes a few patches I had lying around. :)

### CODE REVIEW

File: src/libostree/ostree-fetcher.c [hunk 1]
```diff
@@ -277,6 +277,24 @@ session_thread_config_flags (ThreadClosure *thread_closure,
 }
 
 static void
+on_authenticate (SoupSession *session, SoupMessage *msg, SoupAuth *auth,
+                 gboolean retrying, gpointer user_data)
+{
+  if (msg->status_code == SOUP_STATUS_PROXY_UNAUTHORIZED) {
```
Comment (cgwalters) [inline src/libostree/ostree-fetcher.c]: Brace should be on a newline here.

File: src/libostree/ostree-fetcher.c [hunk 2]
```diff
@@ -277,6 +277,24 @@ session_thread_config_flags (ThreadClosure *thread_closure,
...
+    SoupURI *uri = NULL;
+    g_object_get (session, SOUP_SESSION_PROXY_URI, &uri, NULL);
+    if (retrying)
+      {
+        g_autofree char *s = soup_uri_to_string (uri, FALSE);
+        g_warning ("Invalid username or password for proxy '%s'", s);
```
Comment (cgwalters) [inline src/libostree/ostree-fetcher.c]: I really don't like `g_warning()` as a general rule, particularly for things that can happen "normally" from things like incorrect passwords.

Do we really have no way to propagate this into a proper error back to the caller?  Maybe we have to do something like cache the error and return it later.

File: docs/manual/deployment.md [hunk 3]
```diff
@@ -43,13 +43,14 @@ deployment.  For short, we will call this the "tree", to
...
-the kernel contents; it must be pre-computed before storing the kernel
-in the repository.  Optionally, the tree can contain an initramfs,
-stored as `/boot/initramfs-$checksum`.  If this exists, the checksum
-must include both the kernel and initramfs contents.  OSTree will use
-this to determine which kernels are shared.  The rationale for this is
-to avoid computing checksums on the client by default.
+`/boot/vmlinuz(-.*)?-$checksum`.  The checksum should be a SHA256 hash
+of the kernel contents; it must be pre-computed before storing the
+kernel in the repository.  Optionally, the tree can contain an
+initramfs, stored as `/boot/initramfs(-.*)?-$checksum`.  If this
```
Comment (cgwalters) [inline docs/manual/deployment.md]: While you're amending this, might as well mention all of this content can now be in either `/usr/lib/ostree-boot/` or `/boot`.

**Comment by jlebon** [comment]
Now with less warnings and more errors! ^

**Comment by cgwalters** [comment]
@rh-atomic-bot r+ ba97f3e

**Pull request closed without merging**